internal/services/dataprotection/data_protection_backup_vault_resource.go

```diff
@@ -72,8 +73,35 @@ func resourceDataProtectionBackupVault() *pluginsdk.Resource {
 
 			"identity": commonschema.SystemAssignedIdentityOptional(),
 
+			"retention_duration_in_days": {
+				Type:         pluginsdk.TypeFloat,
+				Optional:     true,
+				ValidateFunc: validation.FloatBetween(14, 180),
+			},
+
+			"soft_delete_setting": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Default:      backupvaults.SoftDeleteStateOn,
+				ValidateFunc: validation.StringInSlice(backupvaults.PossibleValuesForSoftDeleteState(), false),
+			},
+
 			"tags": tags.Schema(),
 		},
+
+		CustomizeDiff: pluginsdk.CustomDiffWithAll(
+			pluginsdk.ForceNewIfChange("soft_delete_setting", func(ctx context.Context, old, new, meta interface{}) bool {
+				return old.(string) == string(backupvaults.SoftDeleteStateAlwaysOn) && new.(string) != string(backupvaults.SoftDeleteStateAlwaysOn)
+			}),
+			pluginsdk.CustomizeDiffShim(func(ctx context.Context, d *pluginsdk.ResourceDiff, v interface{}) error {
+				state := d.Get("soft_delete_setting").(string)
+				_, ok := d.GetOk("retention_duration_in_days")
+				if state == string(backupvaults.SoftDeleteStateOn) && !ok {
+					return fmt.Errorf("`retention_duration_in_days` is required when the `soft_delete_setting` is `On`")
```

Comment:
The way this is implemented will break users as the default is on but `retention_duration_in_days` hasn't been added to the provider yet. So when people update to this version of the provider, Terraform will error with this message. We should probably default `retention_duration_in_days` or remove this check.

Reply:
Hi @mbfrahry thanks a lot. I have updated the code. Could you please take another look?